### PR TITLE
Nationalclouddev

### PR DESCRIPTION
**Pull request by hforeste**
- This fixed the UI navigation issue in applens for national clouds
- Fix detector editing experience in applens for national clouds
- Fix logic bug to detect if `UseAppService` is enabled

### CODE REVIEW

File: ApplensBackend/Services/DiagnosticClientService/DiagnosticClient.cs [hunk 1]
```diff
@@ -56,7 +56,12 @@ public bool IsRunTimeHostEnabled
         {
             get
             {
-                return _configuration.GetValue<bool>("DiagnosticRole:UseAppService");
+                if (bool.TryParse(_configuration["DiagnosticRole:UseAppService"], out bool retVal))
```
Comment (rekhaswaminathan) [inline ApplensBackend/Services/DiagnosticClientService/DiagnosticClient.cs]: Thanks for fixing this!

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts [hunk 2]
```diff
@@ -116,7 +117,8 @@ export class OnboardingFlowComponent implements OnInit {
     this.modalPublishingButtonDisabled = false;
     this.showAlert = false;
 
-    this.userName = this._adalService.userInfo.profile ? this._adalService.userInfo.profile.upn : '';
+    this.userName = Object.keys(this._adalService.userInfo.profile).length > 0 ? this._adalService.userInfo.profile.upn : '';
+    console.log(Object.keys(this._adalService.userInfo.profile).length === 0);
```
Comment (rekhaswaminathan) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts]: do we need console logging here?

Comment (hforeste) [reply AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts]: Nope, removed in new commit

File: AngularApp/projects/applens/src/app/modules/dashboard/dashboard/dashboard.component.ts [hunk 3]
```diff
@@ -75,7 +76,7 @@ export class DashboardComponent implements OnDestroy {
       this._router.navigate([], { queryParams: routeParams, relativeTo: this._activatedRoute });
     }
 
-    if(environment.adal.enabled){
+    if((this.showUserInformation = environment.adal.enabled)){
```
Comment (cindylovescoding) [inline AngularApp/projects/applens/src/app/modules/dashboard/dashboard/dashboard.component.ts]: So the condition when this statement will be be true is whenever the right side "environment.adal.enabled = true" right?

Comment (hforeste) [reply AngularApp/projects/applens/src/app/modules/dashboard/dashboard/dashboard.component.ts]: Yep

**Pull request merged**